server/src/main/java/org/elasticsearch/index/mapper/ConstantFieldType.java

```diff
@@ -79,6 +79,10 @@ private static String valueToString(Object value) {
     public final Query termQuery(Object value, QueryShardContext context) {
         String pattern = valueToString(value);
         if (matches(pattern, context)) {
+            if (context != null && context.getMapperService().hasNested()) {
```

Comment:
Is this for backwards compatibility? Nested fields are defined using a different field in 8x

Reply:
Yes, I wasn't sure about this, there was a test failing without it here: https://github.com/elastic/elasticsearch/blob/master/server/src/test/java/org/elasticsearch/index/mapper/TypeFieldTypeTests.java#L70, so I assumed its still valid. If this is not the case I'm happy to remove this together with the test, but I wasn't sure about the implications.

Reply:
Yes, I don't think this applies any more, as nested queries are always built via the `NestedPathFieldMapper`, so you should be able to remove this block and those few lines of the test.